src/website/app/demos/Menu/examples/Menu/renderItem.js

```diff
@@ -0,0 +1,154 @@
+/* @flow */
+import { withTheme } from 'glamorous';
+import {
+  createStyledComponent,
+  pxToEm
+} from '../../../../../../library/styles';
+import { mineralTheme } from '../../../../../../library/themes';
+import Menu, { MenuItem } from '../../../../../../library/Menu';
+import Avatar from '../../../../../../library/Avatar';
+import DemoLayout from '../../components/DemoLayout';
+import { componentTheme as menuItemTheme } from '../../../../../../library/Menu/MenuItem';
+import renderPropDescription from '../../../shared/renderPropDescription';
+
+export default {
+  id: 'render-item',
+  title: 'Custom Rendering - renderItem',
+  backgroundColor: mineralTheme.color_gray_10,
+  description: `Use the \`renderItem\`
+[render prop](https://reactjs.org/docs/render-props.html) to provide custom
+rendering control of all [MenuItems](/components/menu-item) in the Menu.
+
+${renderPropDescription}
+
+Some things to keep in mind:
+
+1. Remember to include \`:focus\`, \`:hover\`, and \`:active\` styles.
+  - If this is to be used for a [Dropdown](/components/dropdown), also apply your
+     focus/hover styles when \`props.isHighlighted === true\`.
+1. Remember to accommodate the disabled state (\`props.disabled === true\`).
+1. If your app supports RTL languages, you can use \`theme.direction\` to
+   conditionally apply the necessary styles.`,
+  scope: {
+    Avatar,
+    createStyledComponent,
+    DemoLayout,
+    Menu,
+    MenuItem,
+    menuItemTheme,
+    pxToEm,
+    withTheme
+  },
+  source: `
+    () => {
+      const data = [
+        {
+          avatar: '/images/avatar.svg',
+          text: 'Newton',
+          work: 'Principia Mathematica'
+        },
+        {
+          avatar: '/images/avatar.svg',
+          text: 'Descartes',
+          work: 'La Géométrie'
+        },
+        {
+          avatar: '/images/avatar.svg',
+          text: 'Euclid',
+          work: 'Elements'
```

Comment:
I think this may be more clear if the property names aligned; because we're calling the secondary text `work`, can't we use a similar approach to add `text` data as `name`? Also, work as a component confused me a little below before I realized it correlated with the property name `work` since it's such a generic term. Maybe we could rename these properties to `philosopher` & `treatise` to make it extra obvious that this is not how Mineral handles data automatically?

Reply:
I understand what you are after, but I believe that `item.text` must exist in order for it to be reusable for all of the different components and to keep simplified examples.  e.g. The Select trigger relies on `item.text`.  Without `item.text`, the example would also need to implement a custom trigger.